packages/monaco/src/browser/monaco-editor.ts

```diff
@@ -139,6 +141,18 @@ export class MonacoEditor implements TextEditor {
                 event: e.event.browserEvent
             });
         }));
+        this.toDispose.push(codeEditor.onDidScrollChange(e => {
+            this.onScrollChangedEmitter.fire(undefined);
+        }));
+    }
+
+    getVisibleRanges(): Range[] {
+        return this.editor.getVisibleRanges().map(range =>
```

Comment:
```suggestion
        return this.editor.getVisibleRanges().map(range => this.m2p.asRange(change.range));
```

Reply:
Please fix the position mapping: Monaco is 1-based, and LSP (used in TextEditor interface) is 0-based.

Reply:
> synchronization may break if the preview is quickly scrolled all the way up to the top

this might be related, not sure though.

Reply:
@AlexTugarev Ah! I see! Both 'MonacoToProtocolConverter' and 'ProtocolToMonacoConverter' handle position mapping automatically, so the position mapping issue can be fixed by adopting this suggestion. Got a little confused at first ：)